scaleway/resource_k8s_pool_test.go

```diff
@@ -765,91 +779,107 @@ resource "scaleway_k8s_pool" "default" {
 	cni = "calico"
 	version = "%s"
 	tags = [ "terraform-test", "scaleway_k8s_cluster", "minimal" ]
-	delete_additional_resources = true
+	delete_additional_resources = false
+	private_network_id = scaleway_vpc_private_network.minimal.id
 }
 %s`, version, pool)
 }
 
 func testAccCheckScalewayK8SPoolConfigPlacementGroup(version string) string {
 	return fmt.Sprintf(`
 resource "scaleway_instance_placement_group" "placement_group" {
-  name        = "pool-placement-group"
+  name        = "test-pool-placement-group"
   policy_type = "max_availability"
   policy_mode = "optional"
 }
 
 resource "scaleway_k8s_pool" "placement_group" {
-    name               = "placement_group"
+    name               = "test-pool-placement-group"
     cluster_id         = scaleway_k8s_cluster.placement_group.id
     node_type          = "gp1_xs"
     placement_group_id = scaleway_instance_placement_group.placement_group.id
     size               = 1
 }
 
+resource "scaleway_vpc_private_network" "placement_group" {
+	name = "test-pool-placement-group"
+}
+
 resource "scaleway_k8s_cluster" "placement_group" {
-  name	  = "placement_group"
+  name	  = "test-pool-placement-group"
   cni	  = "calico"
   version = "%s"
   tags	  = [ "terraform-test", "scaleway_k8s_cluster", "placement_group" ]
-  delete_additional_resources = true
+  delete_additional_resources = false
+  private_network_id = scaleway_vpc_private_network.placement_group.id
 }`, version)
 }
 
 func testAccCheckScalewayK8SPoolConfigPlacementGroupWithCustomZone(version string) string {
 	return fmt.Sprintf(`
 resource "scaleway_instance_placement_group" "placement_group" {
-  name        = "pool-placement-group"
+  name        = "test-pool-placement-group"
   policy_type = "max_availability"
   policy_mode = "optional"
   zone        = "nl-ams-2"
 }
 
-resource "scaleway_k8s_pool" "pool" {
-  name               = "placement_group"
-  cluster_id         = scaleway_k8s_cluster.cluster.id
+resource "scaleway_k8s_pool" "placement_group_2" {
+  name               = "test-pool-placement-group-2"
+  cluster_id         = scaleway_k8s_cluster.placement_group_2.id
   node_type          = "gp1_xs"
   placement_group_id = scaleway_instance_placement_group.placement_group.id
   size               = 1
-  region             = scaleway_k8s_cluster.cluster.region
+  region             = scaleway_k8s_cluster.placement_group_2.region
   zone               = scaleway_instance_placement_group.placement_group.zone
 }
 
-resource "scaleway_k8s_cluster" "cluster" {
-  name	  = "placement_group"
+resource "scaleway_vpc_private_network" "placement_group" {
+	name = "test-pool-placement-group"
+	region = "nl-ams"
+}
+
+resource "scaleway_k8s_cluster" "placement_group_2" {
+  name	  = "test-pool-placement-group-2"
   cni	  = "calico"
   version = "%s"
   tags	  = [ "terraform-test", "scaleway_k8s_cluster", "placement_group" ]
   region  = "nl-ams"
-  delete_additional_resources = true
+  delete_additional_resources = false
```

Comment:
It's because the cluster gets destroyed and another one is created between step 1 and 2 of the test and if `delete_additional_resources` is set to `true`, the private network gets destroyed with it since it's empty and considered an additional resource of the cluster.

I could also switch back to `true` and create a new private network in the second step if it's better.

I don't think that there were any dangling resources though.

Reply:
Apparently there was no need for this change so I rolled-back.